### ISSUE TITLE
Proof?

### ISSUE THREAD

**Issue by amilopowers** [Zyg0te/Locker#11]
First I have to thank you for your awesome app! Ever since I got rid off my iPhone I wanted to have that on my Android device.

Do you have a video proofing that it actually works? I do not want to try it on my phone... I don't own a development device yet.

**Comment by Zyg0te** [comment]
Hi there and thanks for chiming in.

I haven't made a demonstration video but I have plenty of devices lying around so I could certainly record the app in action.

**Comment by amilopowers** [comment]
Actually I think the AOSP should implement that!

> but I have plenty of devices lying around
That'd be awesome! If you have a spare hammerhead it would be even better ;-)

**Comment by n76** [comment]
> Actually I think the AOSP should implement that!

The framework is in AOSP but the controls are missing. I strongly suspect that is was done that way so that most people would be forced to use Google's Android Device Manager.

**Comment by amilopowers** [comment]
I have another rather technical question.

Does Locker just break the encryption keys or does it really wipe the phone? The iPhone does it the 1st way which is somehow useless.

**Comment by n76** [comment]
> Does Locker just break the encryption keys or does it really wipe the phone? The iPhone does it the 1st way which is somehow useless.

From http://developer.android.com/guide/topics/admin/device-admin.html I think it reboots to recovery and wipes the /data partition. I don't know how long that takes or if it could be stopped before it is complete.

But all it really needs to do is wipe out the key storage at the beginning of the volume: Once it does that everything else cannot be decrypted and is functionally no different than random data.

Which brings up the question: Why do you think that Apple's "breaking the encryption keys" "is somehow useless"? Theoretically that is all that is needed.

**Comment by Zyg0te** [comment]
Correct! The device reboots to recovery mode and re-formats the /data partition. The storage isnt overwritten so its not a secure wipe, which is yet another reason to activate full disk encryption as that will mitigate this weakness.

**Comment by Zyg0te** [comment]
> That'd be awesome! If you have a spare hammerhead it would be even better ;-)

Actually, I do have a Hammerhead available but I was planning on using my Nexus 6, as I have to re-flash it anyways after messing with the sensor HAL too much :)

**Comment by bitterman1970** [comment]
I have personally tried this on my BP1 and can confirm that it does indeed work
